### PR TITLE
Reservations RP 2018-06-01

### PR DESCRIPTION
**Pull request by juhee0202**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservationCatalog.md [hunk 1]
```diff
@@ -13,33 +13,65 @@ Get the catalog of available reservation
 ## SYNTAX
 
 ```
-Get-AzureRmReservationCatalog [-SubscriptionId <String>] [<CommonParameters>]
+Get-AzureRmReservationCatalog [-SubscriptionId <String>] [-ReservedResourceType <String>] [-Location <String>][<CommonParameters>]
 ```
 
 ## DESCRIPTION
-Get the regions and skus that are available for Reserved Instance purchase for the specified Azure subscription.
+Get the regions and skus that are available for RI purchase for the specified Azure subscription.
```
Comment (juhee0202) [inline src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservationCatalog.md]: revert back to Reserved Instance

File: src/ResourceManager/Reservations/Commands.Reservations/help/Update-AzureRmReservation.md [hunk 2]
```diff
@@ -15,13 +15,14 @@ Update a `Reservation`.
...
- [-AppliedScope <String>] [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+ [-AppliedScope <String>] [-InstanceFlexibility <String>] [-DefaultProfile <IAzureContextContainer>] 
+ [-WhatIf] [-Confirm] [<CommonParameters>]
 ```
 
 ### PipeObject
 ```
-Update-AzureRmReservation -AppliedScopeType <String> [-AppliedScope <String>] -Reservation <PSReservation>
- [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+Update-AzureRmReservation -AppliedScopeType <String> [-AppliedScope <String>] [-InstanceFlexibility <String>
```
Comment (juhee0202) [inline src/ResourceManager/Reservations/Commands.Reservations/help/Update-AzureRmReservation.md]: close bracket ]

**Comment by vladimir-shcherbakov** [comment]
Hi @juhee0202 
For the cmdlet  public API changes we require a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr).
Please complete the review to proceed with the PR.

**Comment by juhee0202** [comment]
Link to cmdlet review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/98

**Comment by juhee0202** [comment]
Updated test recordings. All tests passing in Playback mode.

**Comment by juhee0202** [comment]
Hi @maddieclayton, the cmdlet review has been completed.

**Comment by juhee0202** [comment]
the CI build keeps on failing because it is getting the older version of the .NET SDK. What files do I need to update to get the latest version? I already updated some versioning in [*.csproj and packages.config](https://github.com/Azure/azure-powershell/pull/6390/commits/5ae4bad47642ee97f8df7c374010bedea4a0b924).


    Restoring packages for /home/travis/build/Azure/azure-powershell/src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Commands.Resources.Rest.Netcore.csproj...

    Installing Microsoft.Azure.Management.Reservations 1.0.0-preview.

File: src/ResourceManager/Reservations/ChangeLog.md [hunk 3]
```diff
@@ -20,6 +20,14 @@
 -->
 ## Current Release
 
+## Version 0.1.4
```
Comment (juhee0202) [inline src/ResourceManager/Reservations/ChangeLog.md]: todo: update this appropriately and find out if any other places need version change

**Comment by juhee0202** [comment]
@maddieclayton, @cormacpayne The current version of the Reservations powershell module is 0.1.3. The changes in this pull request contain some breaking changes. What would be the most appropriate version for this new powershell module? We would like to stay in preview for now. Would it be 0.1.4?

**Comment by maddieclayton** [comment]
@juhee0202 No need to update the version, we have a script that will update the module based upon the changes made.  But, just for your information, the right bump for a preview module with breaking changes is 0.1.3 -> 0.2.0

**Comment by juhee0202** [comment]
@vladimir-shcherbakov @maddieclayton @cormacpayne may I know who is reviewing my pull request? Also, I am not sure why the build is failing. Any ideas on how to fix this?

16:23:01 Started calculate disk usage of build
16:23:01 Finished Calculation of disk usage of build in 0 seconds
16:23:01 ERROR: ADXWorker59s is offline; cannot locate JDK-9.0.4
16:23:01 Setting status of 44d4d0fd7f021dffa2efbdb74329661bd7b4a584 to FAILURE with url https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/5314/ and message: 'Build finished. '
16:23:01 ERROR: ADXWorker59s is offline; cannot locate JDK-9.0.4

File: src/ResourceManager/Reservations/ChangeLog.md (hunk 3 above)
Comment (juhee0202) [reply src/ResourceManager/Reservations/ChangeLog.md]: move this to 'Current Release' section

**Comment by juhee0202** [comment]
@cormacpayne I am getting StaticAnalysis errors during build. How can I fix this?

16:25:30 c:\workspace\powershell\build.proj(502,5): error MSB3073: The command "c:\workspace\powershell\src\Package\StaticAnalysis.exe c:\workspace\powershell\src\Package\Debug c:\workspace\powershell\src\Package true false AzureRM.Reservations" exited with code 255.
16:25:30 c:\workspace\powershell\build.proj(508,5): error : StaticAnalysis has failed.  Please follow the instructions on this doc: https://github.com/Azure/azure-powershell/blob/preview/documentation/Debugging-StaticAnalysis-Errors.md

**Comment by maddieclayton** [comment]
@juhee0202  Can you please fix the merge conflict? We have moved ChangeLog/psd1 files from src/ResourceManager/{Module} to src/ResourceManager/{Module}/Commands.{Module}, so you will need to move your changelog/psd1 updates there.

**Comment by juhee0202** [comment]
@maddieclayton the merge conflict has been resolved by moving the ChangeLog to the folder you specified above.

**Comment by vladimir-shcherbakov** [comment]
ci-build failed.

**Comment by juhee0202** [comment]
@vladimir-shcherbakov fixed the ci-build

File: tools/StaticAnalysis/Exceptions/AzureRM.Reservations/SignatureIssues.csv [hunk 4]
```diff
@@ -1 +1,15 @@
 "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
+"c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll","Microsoft.Azure.Commands.Network.GetAzureRMExpressRouteCrossConnectionArpTableCommand","Get-AzureRMExpressRouteCrossConnectionArpTable","1","8510","Cmdlet 'Get-AzureRMExpressRouteCrossConnectionArpTable' has multiple parameter sets, but no defined default parameter set.","Define a default parameter set in the cmdlet attribute."
```
Comment (cormacpayne) [inline tools/StaticAnalysis/Exceptions/AzureRM.Reservations/SignatureIssues.csv]: @juhee0202 would you mind moving all of these Network exceptions into `tools/StaticAnalysis/Exceptions/AzureRM.Network/SignatureIssues.csv`? This makes it easier to track suppressed exceptions per RP

Comment (juhee0202) [reply tools/StaticAnalysis/Exceptions/AzureRM.Reservations/SignatureIssues.csv]: will do

**Pull request merged**